contracts/src/Assets.sol

```diff
@@ -174,4 +176,80 @@ library Assets {
 
         emit IGateway.TokenRegistrationSent(token);
     }
+
+    // @dev Register a new fungible Polkadot token for an agent
+    function registerTokenByID(bytes32 tokenID, address token, bytes32 agentID) external {
+        AssetsStorage.Layout storage $ = AssetsStorage.layout();
+        if ($.tokenRegistryByID[tokenID].isRegistered == true) {
+            revert TokenAlreadyRegistered();
+        }
+        TokenInfo memory info =
+            TokenInfo({isRegistered: true, isForeign: true, tokenID: tokenID, agentID: agentID, token: token});
+        $.tokenRegistry[token] = info;
+        $.tokenRegistryByID[tokenID] = info;
+    }
+
+    // @dev Get token address by tokenID
+    function getTokenInfo(bytes32 tokenID) external view returns (TokenInfo memory) {
+        AssetsStorage.Layout storage $ = AssetsStorage.layout();
+        if ($.tokenRegistryByID[tokenID].isRegistered == false) {
+            revert TokenNotRegistered();
+        }
+        return $.tokenRegistryByID[tokenID];
+    }
+
+    // @dev Transfer polkadot native tokens back
+    function sendForeignToken(
+        address executor,
+        TokenInfo memory info,
+        address sender,
+        ParaID destinationChain,
+        MultiAddress calldata destinationAddress,
+        uint128 destinationChainFee,
+        uint128 amount
+    ) external returns (Ticket memory ticket) {
+        CoreStorage.Layout storage $core = CoreStorage.layout();
+
+        address agent = $core.agents[info.agentID];
+        if (agent == address(0)) {
+            revert AgentDoesNotExist();
+        }
```

Comment:
Rather move this to `Gateway.sendToken` and pass `agent` as a parameter.

The `Assets` library doesn't need to know how agents are stored.

Reply:
https://github.com/Snowfork/snowbridge/pull/1155/commits/edccccdf014c5d4a8934591bef649a870e430992